src/couch/src/couch_doc.erl

```diff
@@ -191,6 +198,15 @@ parse_revs(_) ->
     throw({bad_request, "Invalid list of revisions"}).
 
 
+validate_docid(DocId, DbName) ->
+    case DbName =:= ?l2b(config:get("mem3", "shards_db", "_dbs")) andalso
+        lists:member(DocId, ?SYSTEM_DATABASES) of
```

Comment:
assert in the function head that DocId is binary, just in case.

Reply:
as we discussed on IRC, `lists:member` can handle a list being passed in as DocId, so we'll leave this be for now.

```
1> Foo = [["a", "b"], "c", "d"].
[["a","b"],"c","d"]
2> lists:member(["a", "b"], Foo).
true
```